File: two_pointers/remove_duplicates.js
const remove_duplicates = function(arr) {
  let left = 0;

  for (let right = 1; right < arr.length; right++) {
    if (arr[left] !== arr[right]) {
      left++;
      arr[left] = arr[right];
    }
  }

  return left + 1;
};

File: two-pointers/same_and_opposite/remove_duplicates.py
def remove_duplicates(arr):
  left, right = 0, 1

  while right < len(arr):
    if arr[left] != arr[right]:
      left += 1
      arr[left] = arr[right]

    right += 1

  return left + 1

# Time Complexity: O(n) bc we make one pass
# Space Complexity: O(1)